src/Power.cpp

```diff
@@ -413,35 +427,80 @@ class AnalogBatteryLevel : public HasBatteryLevel
 #endif
 
 #if HAS_TELEMETRY && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR && !defined(ARCH_PORTDUINO)
+    int16_t getINACurrent()
+    {
+        const uint32_t ina_addr = config.power.device_battery_ina_address;
+        int16_t ret;
+
+        if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA219].first == ina_addr) {
+            ret = ina219Sensor.getCurrentMa();
+
+        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA260].first == ina_addr) {
+            ret = ina260Sensor.getCurrentMa();
+
+        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA3221].first == ina_addr) {
+            ret = ina3221Sensor.getCurrentMa();
+
+        /* don't know how to read current, return -1mA (negative so isCharging() will do the right thing) */
+        } else {
+            ret = -1;
+        }
+
+        return ret;
+    }
+
     uint16_t getINAVoltage()
     {
-        if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA219].first == config.power.device_battery_ina_address) {
-            return ina219Sensor.getBusVoltageMv();
-        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA260].first ==
-                   config.power.device_battery_ina_address) {
-            return ina260Sensor.getBusVoltageMv();
-        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA3221].first ==
-                   config.power.device_battery_ina_address) {
-            return ina3221Sensor.getBusVoltageMv();
+        const uint32_t ina_addr = config.power.device_battery_ina_address;
+        uint16_t ret;
+
+        if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA219].first == ina_addr) {
+            ret = ina219Sensor.getBusVoltageMv();
+
+        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA260].first == ina_addr) {
+            ret = ina260Sensor.getBusVoltageMv();
+
+        } else if (nodeTelemetrySensorsMap[meshtastic_TelemetrySensorType_INA3221].first == ina_addr) {
+            ret = ina3221Sensor.getBusVoltageMv();
+
+        /* don't know how to read voltage, return 0mV */
+        } else {
+            ret = 0;
         }
-        return 0;
+
+        return ret;
```

Comment:
I always get the sense that maybe it's good to avoid reworking stuff like this just for the sake of it, to keep the diff down, unless the PR is explicitly tidying or something, but I'm not sure if there's an official stance on it.

Reply:
I was trying to get rid of the line wrapping; we've got some insanely long variable names that blow out the lines and then the wrapping makes everything harder to read. I'm trying to not change much but the voltage function was almost identical and immediately below the one I added.

Reply:
That's a pretty valid reason 👍